res/skins/LateNight/samplers_rack.xml

```diff
@@ -1,43 +1,68 @@
 <Template>
+
   <WidgetGroup>
     <Layout>horizontal</Layout>
     <SizePolicy>me,max</SizePolicy>
+    <Connection>
+      <ConfigKey>[Samplers],show_samplers</ConfigKey>
+      <BindProperty>visible</BindProperty>
+    </Connection>
     <Children>
 
-      <Template src="skin:sampler.xml">
-        <SetVariable name="samplernum">1</SetVariable>
-      </Template>
-
-      <WidgetGroup><Size>10f,1min</Size></WidgetGroup>
-
-      <Template src="skin:sampler.xml">
-        <SetVariable name="samplernum">2</SetVariable>
-      </Template>
-
-      <WidgetGroup><Size>10f,1min</Size></WidgetGroup>
+      <WidgetGroup>
+      <Layout>horizontal</Layout>
+        <Children>
+          <!-- Select between 1-4 big or orthers -->
+          <WidgetStack currentpage="[Master],show_4sampler">
+            <Children>
 
-      <Template src="skin:sampler.xml">
-        <SetVariable name="samplernum">3</SetVariable>
-      </Template>
+             <!-- Select between 5-8 big or 8 tiny -->
+                  <WidgetStack currentpage="[Samplers],show_big">
```

Comment:
[Skin],expand_samplers

Reply:
please keep an eye on the indentations, and please make the comments match the actual widgets